mne/io/edf/edf.py

```diff
@@ -621,13 +621,13 @@ def _read_edf_header(fname, exclude):
         ch_names = _unique_channel_names(ch_names)
         orig_units = dict(zip(ch_names, units))
 
-        physical_min = np.array([float(fid.read(8).decode())
+        physical_min = np.array([float(fid.read(8).decode().rstrip('\x00'))
```

Comment:
I would do a `.split('\x00')[0]` instead. Really I think this is an old-school C-string null termination, i.e., read until 8 bytes or `\x00` is encountered, and `\x00` replicates this behavior.

My C is rusty, though, so take this with a grain of salt and if someone knows better feel free to correct me...

Reply:
... really I would refactor `_edf_str_int` to be `_edf_str_cast` and do a `float` cast here. It uses the `.split` method and keeps the code more DRY